packages/payload/src/uploads/generateFileData.ts

```diff
@@ -85,14 +85,18 @@ export const generateFileData = async <T>({
         const filePath = `${staticPath}/${filename}`
         const response = await getFileByPath(filePath)
         file = response
-        overwriteExistingFiles = true
+        if (operation !== 'duplicate') {
+          overwriteExistingFiles = true
+        }
```

Comment:
@DanRibbens see here, this code executes on both `update` and `create` operations and will lead to the unique file name error, that's why I introduced it.

I'm already passing `overwriteExistingFiles` from the caller, but this overwrites it.